### PR TITLE
PXC-3567 Merge codership/mysql-wsrep 5.7.32 -> PXC 5.7

### PR DESCRIPTION
**Pull request by venkatesh-prasad-v**
https://jira.percona.com/browse/PXC-3567

Merge conflicts resolved

***
Updated the WSREP_PATCH_VERSION to 49.

***
Updated WSREP-REVISION to reflect the upstream refresh revision.
(commit 1e88d72).

***
Reverted upstream fix for
"Fix test of '-k' option for timeout(1) in wsrep_sst_xtrabackup-v2"
(commit 291cd9f and 1b073cd)
in favour of
"Fix error handling when donor is not able to serve SST"
https://jira.percona.com/browse/PXC-3424
(commit eddd130 and 647e728)

***
Corrected the fix for the bug
"Xid events for LOAD DATA splitted transactions"
(commit 660b5f4)
to include the check in both `wsrep_replicate()` and
`wsrep_run_wsrep_commit()` hooks and to fix the test failure of
galera.galera_load_data testcase.

***
Corrected the filtering logic in `check_table_map()` introduced by
"Replication filters should not affect galera cluster events"
(commit cf3987b)
to fix the test failure of galera.galera_repl_filtersA and
galera.galera_repl_filtersB test cases.

***
Adjusted the test galera.mysql-wsrep#372.test to

1. Replace occurences of `SET SESSION wsrep_on=ON/OFF` into
`SET GLOBAL wsrep_on=ON/OFF`, as `wsrep_on` is a session variable in PXC.

2. Add a suppression to the expected warnings.

***
Adjusted the test galera.galera_UK_conflict.test to use the correct
thread state `wsrep: replaying write set(..)`.

***
Re-recorded the following test cases.

1. main.mysqld--help-notwin
2. galera.galera_defaults
3. galera.galera_load_data
4. perfschema.show_sanity
5. suite.all_vars

Testing Done
---
**Round 1 (01-Feb-21)**
1. Galera and sysvar suite: https://pxc.cd.percona.com/view/PXC%205.7/job/pxc-5.7-param/101/testReport/
Parameters: https://pxc.cd.percona.com/view/PXC%205.7/job/pxc-5.7-param/101/parameters/
```
Failing Tests: 

1. galera.galera_sst_xtrabackup-v2_encrypt_with_key - Fails due to PXC-3508
2. galera.galera_sst_xtrabackup-v2_keyring - Fails due to PXC-3508
```

2. Full MTR: https://pxc.cd.percona.com/view/PXC%205.7/job/pxc-5.7-param/102/console
Parameters: https://pxc.cd.percona.com/view/PXC%205.7/job/pxc-5.7-param/102/parameters/
```
Failing Tests:

1. binlog.binlog_query_filter_rules - Fixed now
2. keyring_vault tests: https://jira.percona.com/browse/PXC-3297
3. main.merge_mmap & main.merge: https://jira.percona.com/browse/PXC-3296
4. main.commit_1innodb: Sporadic failure due to improper cleanup by its previous test main.bug93770. https://jira.percona.com/browse/PS-7479
5. galera.galera_sst_xtrabackup-v2_encrypt_with_key - Fails due to PXC-3508 
6. galera.galera_sst_xtrabackup-v2_keyring - Fails due to PXC-3508
```
**Round 2 (02-Feb-21)**
1. Full MTR: https://pxc.cd.percona.com/view/PXC%205.7/job/pxc-5.7-param/105/console
Parameters: https://pxc.cd.percona.com/view/PXC%205.7/job/pxc-5.7-param/105/parameters/
```
Failing Tests:
1. keyring_vault tests: https://jira.percona.com/browse/PXC-3297
2. main.merge_mmap & main.merge: https://jira.percona.com/browse/PXC-3296
3. main.commit_1innodb: Sporadic failure due to improper cleanup by its previous test main.bug93770. https://jira.percona.com/browse/PS-7479
4. galera.galera_sst_xtrabackup-v2_encrypt_with_key - Fails due to PXC-3508 
5. galera.galera_sst_xtrabackup-v2_keyring - Fails due to PXC-3508
```

### CODE REVIEW

File: mysql-test/suite/galera/t/galera_UK_conflict.test [hunk 1]
```diff
@@ -91,7 +91,8 @@ SET GLOBAL debug = NULL;
 SET debug_sync='RESET';
 
 # wait for BF abort to happen and replaying begin
---let $wait_condition = SELECT COUNT(*)=1 FROM INFORMATION_SCHEMA.PROCESSLIST WHERE STATE= 'wsrep replaying trx' 
+
```
Comment (kamil-holubicki) [inline mysql-test/suite/galera/t/galera_UK_conflict.test]: extra empty line

File: storage/innobase/handler/ha_innodb.cc [hunk 2]
```diff
@@ -9490,13 +9385,14 @@ ha_innobase::write_row(
 
 		m_num_write_row = 0;
 
-		switch (
-		    wsrep_run_wsrep_commit(m_user_thd, wsrep_hton, true)) {
+		int ret=0;
```
Comment (kamil-holubicki) [inline storage/innobase/handler/ha_innodb.cc]: Wouldn't the following be more readable?
```
int ret = foo();
switch(ret) { ...
```

**Review by kamil-holubicki — COMMENTED** [review]
Looks good. 
2 comments/questions

File: storage/innobase/handler/ha_innodb.cc (hunk 2 above)
Comment (venkatesh-prasad-v) [reply storage/innobase/handler/ha_innodb.cc]: Agreed. Done.

File: mysql-test/suite/galera/t/galera_UK_conflict.test (hunk 1 above)
Comment (venkatesh-prasad-v) [reply mysql-test/suite/galera/t/galera_UK_conflict.test]: fixed.

**Review by kamil-holubicki — APPROVED** [review]
lgtm

**Pull request merged**